app/resources/cdl/charge_manager.rb

```diff
@@ -84,8 +58,47 @@ def create_hold(netid:)
       change_set_persister.save(change_set: change_set)
     end
 
-    def available_item_id
-      (item_ids - resource_charge_list.charged_items.map(&:item_id)).first
+    def activate_holds!
+      @resource_charge_list = expire_holds!
+      return unless available_charge_slot? && resource_charge_list.pending_or_active_holds.present?
+      return unless activate_hold_count.positive?
+      activated_holds = Array.new(activate_hold_count).map do
+        hold = resource_charge_list.pending_holds.first
+        hold.expiration_time = 1.hour.from_now
+        hold
+      end
+      change_set = CDL::ResourceChargeListChangeSet.new(resource_charge_list)
+      change_set.validate(hold_queue: resource_charge_list.hold_queue)
+      change_set_persister.save(change_set: change_set)
+      activated_holds.each do |hold|
+        notify_hold_active(hold: hold)
+      end
+    end
+
+    def activate_hold_count
```

Comment:
Is there a better name for this? It seems like something is going to get activated when it's called.

Reply:
It seems to be a count of items available to hold. Not sure how to phrase this more clearly.